File: app/src/main/java/com/sakuna63/tumbin/data/model/Blog.java
package com.sakuna63.tumbin.data.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import io.realm.RealmList;
import io.realm.RealmObject;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Blog extends RealmObject {

    public String name;
    @JsonProperty("active")
    public boolean isActive;
    public String title;
    public String description;
    @JsonProperty("total_posts")
    public int numOfPosts;
    public String url;
    @JsonProperty("updated")
    public int updatedAtInMills;
    public String uuid;
    public String key;
    public Theme theme;
    @JsonProperty("avatar")
    public RealmList<Avatar> avatars;
    public boolean canMessage;
    public boolean shareLikes;
    public boolean shareFollowing;
    public boolean isNsfw; // = following any blog or not
    public boolean followed;
    @JsonProperty("likes")
    public int numOfLikes;
}

File: app/src/main/java/com/sakuna63/tumbin/data/model/Blog.kt
package com.sakuna63.tumbin.data.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import io.realm.RealmList
import io.realm.RealmObject

@JsonIgnoreProperties(ignoreUnknown = true)
open class Blog : RealmObject() {

    lateinit var name: String
    var active: Boolean = false
    var title: String? = null
    var description: String? = null
    var url: String? = null
    var updated: Int = 0
    var uuid: String? = null
    var key: String? = null
//    lateinit var theme: Theme
    @JsonProperty("avatar")
    var avatars: RealmList<Avatar>? = null
    var canMessage: Boolean = false
    var shareLikes: Boolean = false
    var shareFollowing: Boolean = false
    var canBeFollowed: Boolean = false
    var isNsfw: Boolean = false
}
